src/viewProviders/schemas.test.ts

```diff
@@ -60,31 +56,31 @@ describe("SchemasViewProvider search behavior", () => {
   });
 
   beforeEach(() => {
-    sandbox = sinon.createSandbox();
+    provider = SchemasViewProvider.getInstance();
 
-    // stub loader method for fetching schemas
-    ccloudLoader = CCloudResourceLoader.getInstance();
-    sandbox
-      .stub(ccloudLoader, "getSubjects")
-      // three sample schema versions across three subjects
-      .resolves([TEST_CCLOUD_SUBJECT, TEST_CCLOUD_SUBJECT2, TEST_CCLOUD_SUBJECT3]);
+    // wire up the initial test data subjects.
+    const fakeSubjectMap: Map<string, Subject> = new Map();
+    fakeSubjectMap.set(TEST_CCLOUD_SUBJECT.name, TEST_CCLOUD_SUBJECT);
+    fakeSubjectMap.set(TEST_CCLOUD_SUBJECT2.name, TEST_CCLOUD_SUBJECT2);
+    fakeSubjectMap.set(TEST_CCLOUD_SUBJECT3.name, TEST_CCLOUD_SUBJECT3);
+
+    // rewrite private map with fixed value
```

Comment:
Filtering in getChildren(I happen against the keys in `this.subjectsInTreeView`, so adjust the test to preset those instead of route results.

And then adjust the tests to call `provider.setSearch()` directly, avoiding the call to `this.refresh()` done by `schemaSearchSet` event listener, which completely undoes the test setup work we do here.